api/turing/cluster/controller.go

```diff
@@ -198,6 +201,17 @@ func InitClusterControllers(
 		controllers[envName] = ctl
 	}
 
+	selfCtl, err := newController(
+		clusterConfig{
+			ClusterName:     LocalClusterName,
+			InClusterConfig: true,
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to initialize cluster controller")
+	}
+	controllers[LocalClusterName] = selfCtl
```

Comment:
This new controller allows the Turing API to work directly with the cluster in which it is deployed.

Reply:
What do you think about letting the experiment engine declare the service account key file's path relative to the API, rather than the name of the secret in the API's cluster? We can use the `extraVolumes` and `extraVolumeMounts` fields in the chart to mount the engine's SA data: https://github.com/caraml-dev/turing/blob/main/infra/charts/turing/values.yaml#L85

I think it can help with 2 things:
* Makes the implementation simpler (no need to have the local controller, get the secret from the Turing app's cluster)
* Makes local development consistent - this is something I hadn't considered earlier but with the current implementation, if we were to run the API locally and point to a remote cluster for deploying the routers, this step of "getting a secret from the app's cluster" doesn't seem to fit in. :/ 

Wdyt ?

Reply:
Yep that makes sense! On the other hand, if we were to make users specify a `ServiceAccountKeyFilePath` under the experiment engine, the user would now have to potentially specify values in 2 different locations, 1 for the actual `ServiceAccountKeyFilePath` found under `experimentEngines`
```yaml
experimentEngines:
- name: xp
   type: rpc-plugin
   serviceAccountKeyFilePath: /etc/plugins/xp/gcp_service_account/service-account.json
   ...
```
and 1 other for determining which secret (the secret name) to be mounted as a volume in the Turing API pod
```yaml
extraVolumes:
- name: xp-plugin-service-account
  secret:
    secretName: xp-secret
    items:
    - key: service-account.json
       path: service-account.json
```
All other values/configurations can kind of be automatically generated using helm templates automatically. I was thinking if we can go further and make the file path of the service account unconfigurable (where it gets mounted will be determined by the helm template) and allow the users to directly state the name of the secret under the experiment engine. This is pretty much identical to how the original implementation of this PR would require users to specify the `experimentEngines` configuration:
```yaml
experimentEngines:
- name: xp
   type: rpc-plugin
   secretName: xp-secret
   ...
```
We'll then rely on the helm template to automatically generate the volumes and volume mounts and also set the `ServiceAccountKeyFilePath` field under `RouterDefaults.ExperimentEnginePlugins`. What do you think? 🤔

Reply:
The user also needs to set the `extraVolumes `'s `key: service-account.json` to identify what key to look for in the secret, right? Automatically appending to the `extraVolumes` and `extraVolumeMounts` via the Helm templates makes sense to me.

Reply:
Oh yea, that too. So I guess we'll have 2 fields, one for the `secretName` and `key`. But now as I'm thinking through this, I'm starting to realise that this automatic appending of the `extraVolumes` and `extraVolumeMounts` for the experiment engine plugin might be taking things a little too far: we aren't even doing this for the regular service account keys that the Turing API requires (one has to specify the `extraVolumes`, `extraVolumeMounts` and `extraEnvs` manually to ensure the creds file can be properly located). 

For consistency with how we're expecting users to mount the creds file themselves for the Turing API (do-it-yourself, unknown to the Helm charts, secret-agnostic in some sense since we don't care how the creds file get inserted into the pod as long as it's there), I was thinking maybe the best idea is to just keep the `ServiceAccountKeyFilePath` (sticking to your original suggestion) and assume that the user knows how to get the creds file for the plugin into the Turing API's pod somehow. 

Sorry for flip flopping on this 😵‍💫